pkg/reporting/reporting.go

```diff
@@ -165,6 +166,11 @@ func Generate(cfg *config.Config, results *results.ResultsServer, l log.Logger)
 			fmt.Fprint(os.Stderr, string(str))
 		} else {
 			f, err := os.OpenFile(outputFile, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o644)
+			if os.IsNotExist(err) {
+				dir := filepath.Dir(outputFile)
+				os.MkdirAll(dir, 0o744)
+				f, err = os.OpenFile(outputFile, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o644)
```

Comment:
To avoid duplicating this line and also leaving the error from the line above uncaught, I would just call `os.MkdirAll` regardless before even trying to open the file. If the directory exists it will just do [nothing](https://pkg.go.dev/os#MkdirAll).

Reply:
I apply the changes